### PR TITLE
Chore/fix other typos

### PR DESCRIPTION
**Pull request by hoangdv2429**
This PR will fix everything else in addition to #1805

---

#### PR checklist

- [ ] Tests written/updated
- [ ] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [x] Updated relevant documentation (`docs/` or `spec/`) and code comments

### CODE REVIEW

File: docs/architecture/tendermint-core/adr-020-block-size.md [hunk 1]
```diff
@@ -26,7 +26,7 @@ We could also consider using a MaxDataBytes instead of or in addition to MaxByte
 ## MaxBytes vs MaxDataBytes
 
 The [PR #3045](https://github.com/tendermint/tendermint/pull/3045) suggested
-additional clarity/justification was necessary here, wither respect to the use
+additional clarity/justification was necessary here, whether respect to the use
```
Comment (thanethomson) [inline docs/architecture/tendermint-core/adr-020-block-size.md:29]:
```suggestion
additional clarity/justification was necessary here, with respect to the use
```

File: libs/json/encoder.go [hunk 2]
```diff
@@ -28,7 +28,7 @@ func Marshal(v interface{}) ([]byte, error) {
 	return buf.Bytes(), nil
 }
 
-// MarshalIndent marshals the value as JSON, using the given prefix and indentation.
+// MarshalIndent marshals the value as JSON, using the given prefix and indention.
```
Comment (thanethomson) [inline libs/json/encoder.go:31]:
```suggestion
// MarshalIndent marshals the value as JSON, using the given prefix and indentation.
```

Incorrect detection of a spelling error.

File: privval/msgs_test.go [hunk 3]
```diff
@@ -65,7 +65,7 @@ func TestPrivvalVectors(t *testing.T) {
 	proposal := exampleProposal()
 	proposalpb := proposal.ToProto()
 
-	// Create a Reuseable remote error
+	// Create a Reusable remote error
```
Comment (thanethomson) [inline privval/msgs_test.go:68]:
```suggestion
	// Create a reusable remote error
```

**Review by thanethomson — COMMENTED** [review]
Thanks @hoangdv2429! Just a few minor corrections.

**Comment by hoangdv2429** [comment]
@thanethomson I updated it. Thank you!

**Review by thanethomson — APPROVED** [review]

**Pull request merged**